Blocks/HTTP/impl/posix_server.h

```diff
@@ -88,33 +88,48 @@ struct ServeStaticFilesFromCannotServeMoreThanOneIndexFile : ServeStaticFilesExc
 };
 
 struct ServeStaticFilesFromOptions {
-  std::string url_base;
+  // HTTP server route prefix.
+  std::string public_route_prefix;
+
+  // User-facing URL prefix. Used to prefix the route in the trailing slash redirects.
+  std::string public_url_prefix;
```

Comment:
Not in every case, that's why this PR. Imagine: user-facing URL is `http://api.example.com/files/*` but the server serves at `/static/*`. If the server has to issue a trailing slash redirect for `http://api.example.com/files/somedir`, it should return either `/files/somedir/` or `http://api.example.com/files/somedir/`, neither of these contain `/static`